src/main/java/com/iheartradio/m3u8/data/Playlist.java

```diff
@@ -38,13 +45,39 @@ public boolean isExtended() {
     public int getCompatibilityVersion() {
         return mCompatibilityVersion;
     }
+   
+    public boolean hasUnknownTags() {
+        return mUnknownTags.size() > 0;
+    }
+    
+    public List<String> getUnknownTags() {
+        return mUnknownTags;
+    }
+
+    public Builder buildUpon() {
+        return new Builder(mMasterPlaylist, mMediaPlaylist, mUnknownTags, mIsExtended, mIsIframesOnly, mCompatibilityVersion);
+    }
 
     public static class Builder {
         private boolean mIsExtended;
+        private boolean mIsIframesOnly;
         private MasterPlaylist mMasterPlaylist;
         private MediaPlaylist mMediaPlaylist;
+        private List<String> mUnknownTags = Collections.emptyList();
```

Comment:
The `Playlist` is similar to a C style union of a `MasterPlaylist` and `MediaPlaylist` in that it has one or the other but not both. As such, it would be better for the unknown tags to live in the contained playlist (master or media) which contains the actual playlist information. This is to maintain the focus of a `Playlist` as a mechanism for figuring out which type of playlist has been parsed.